File: Ordenacao1/cpp/src/selection_sort.cpp
#include "selection_sort.h"

#include "utils.h"

bool selection_sort::selection_sort(std::vector<int> &vec)
{

  int i, j, imin;

  for(i=0; i < vec.size(); i++) {

    imin = i;

    for(j=i; j < vec.size(); j++) {

      if(vec[j] < vec[imin]) {
        imin = j;
      }

    }

    if(i != imin) {
      swap(vec, i, imin);
    }

  }

  return true;

}

File: Ordenacao1/rust/src/selection_sort.rs

pub fn sort(vec: &mut Vec<usize>) {

    let vlen = vec.len();

    for i in 0..vlen {

        let min = (i..vlen).min_by_key(|x| vec[*x]).unwrap();
        vec.swap(min, i);

    }

}
